pkg/reconciler/pipelinerun/resources/resultrefresolution.go

```diff
@@ -250,30 +187,6 @@ func getReferencedTaskRun(pipelineState PipelineRunState, reference *v1beta1.Res
 	return referencedPipelineTask.TaskRun, nil
 }
 
-func getTaskRunStatus(pipelineStatus v1beta1.PipelineRunStatus, pipelineTaskName string) (*v1beta1.TaskRunStatus, string, error) {
-	for key, taskRun := range pipelineStatus.PipelineRunStatusFields.TaskRuns {
-		// check if the task run was successful
-		if taskRun.PipelineTaskName == pipelineTaskName {
-			c := taskRun.Status.GetCondition(apis.ConditionSucceeded)
-			if c == nil {
-				return nil, "", fmt.Errorf("could not find a successful task run status for task %q referenced by result", pipelineTaskName)
-			}
-			return taskRun.Status, key, nil
-		}
-	}
-	return nil, "", fmt.Errorf("could not find task run status for task %q referenced by result", pipelineTaskName)
-}
-
-func findTaskResultForPipelineResult(taskStatus *v1beta1.TaskRunStatus, reference *v1beta1.ResultRef) (*v1beta1.TaskRunResult, error) {
-	results := taskStatus.TaskRunStatusFields.TaskRunResults
-	for _, result := range results {
-		if result.Name == reference.Result {
-			return &result, nil
-		}
-	}
-	return nil, fmt.Errorf("Could not find result with name %s for task run %s", reference.Result, reference.PipelineTask)
```

Comment:
This `error` is getting lost and not surfaced to user or logs,  here is the chain of function calls: `findTaskResultForPipelineResult`  <- `resolveResultRefForPipelineResult`  <- `extractResultRefsForPipelineResults` <- `convertPipelineResultToResultRefs`

 `convertPipelineResultToResultRefs` returns nil if `error` is not `nil` without logging it or returning the error along but not sure how the result reference as is makes it to the `pipelineResults`. 

With, #3499, will able to pick up such errors and report them.

Reply:
Good catch, this is the reason that I ultimately decided the new `taskResultValue()` func shouldn't bother returning an error and just a pointer to the string value if there is one.